Adds documentation about git configuration (#3672)
Codecademy/docs: content/git/concepts/config/config.md

---

> title: Git Config

Reviewer: This metadata section is incorrect please see look at another similar term entry or a template like the example linked - 
https://github.com/Codecademy/docs/blob/main/documentation/term-entry-template.md?plain=1

---

# Git Config

> **Git configuration** is a fundamental aspect of using Git, allowing users to customize various settings and preferences on both a global and repository-specific level. Configuration settings in Git help you manage aspects such as your identity, preferred text editor, merge strategies, and much more. Git stores these settings in configuration files, and understanding how to work with these configurations is essential for efficient and smooth collaboration with Git.

Reviewer: ```suggestion
**Git configuration** is a fundamental aspect of using Git, allowing users to customize various settings and preferences on both a global and repository-specific level. Configuration settings help users manage various aspects of the profile and operation of Git including: identity details, preferred text editor, merge strategies, and much more. Git stores these settings in configuration files, and understanding how to work with these configurations is essential for efficient and smooth collaboration with Git.
```

Reviewer: docs avoids the use of first and second person pronouns ( I you we). Please update the rest of the document accordingly.

## Viewing Git Configurations

You can view your Git configuration using the `git config` command. To see your global configuration settings, use:

> ```bash

Reviewer: ```suggestion
```shell
```

git config --global --list
```

## Setting Git Configurations 

You can set your Git configurations using the git config command. For example, to set your name and email address, which are crucial for authorship identification:

```bash
git config --global user.name "Your Name"
git config --global user.email "your.email@example.com"
```

> You can also set repository-specific configurations without the --global flag.

Reviewer: ```suggestion
Repository-specific configurations can also be set without the `--global flag`.
```

## Removing Git Configurations

> If you need to remove a specific configuration setting, use the --unset flag with the git config command. For example, to remove the user email from your global configuration:

Reviewer: ```suggestion
To remove a specific configuration setting, use the `--unset` flag with the `git config` command. For example, to remove the user email from your global configuration:
```

Author: I've revised the document again. Please review it, and if it's not good enough, feel free to let me know, and I'll make further improvements.
@caupolicandiaz

---

> Title: 'Git Config'

Reviewer 2: ```suggestion
Title: 'Config'
```

> Description: 'Learn how to configure Git settings for smooth collaboration.'

Reviewer 2: ```suggestion
Description: 'Git configuration is a fundamental aspect of using Git, allowing users to customize various settings and preferences on both a global and repository-specific level.'
```

Subjects:
  - 'Bash/Shell'
  - 'Developer Tools'
Tags:
  - 'Git'
  - 'Version Control'
CatalogContent:
  - 'learn-git'
  - 'learn-the-command-line'
---

**Git configuration** is a fundamental aspect of using Git, allowing users to customize various settings and preferences on both a global and repository-specific level. Configuration settings manage various aspects of the profile and operation of Git, including identity details, preferred text editors, merge strategies, and more. Git stores these settings in configuration files, making it essential for efficient and smooth collaboration with Git.

## Viewing Git Configurations

To view Git a configuration, utilize the `git config` command. To see global configuration settings, use:

```shell
git config --global --list
```

## Setting Git Configurations

Git configurations can be set using the `git config` command. For instance, to set name and email address, which are crucial for authorship identification:

```shell
git config --global user.name "Your Name"
git config --global user.email "your.email@example.com"
```

> Repository-specific configurations can also be set without the `--global flag`.

Reviewer 2: ```suggestion
Repository-specific configurations can also be set without the `--global` flag.
```

## Removing Git Configurations

> To remove a specific configuration setting, use the `--unset` flag with the `git config` command. For example, to remove the user email from your global configuration:a

Reviewer 2: ```suggestion
To remove a specific configuration setting, use the `--unset` flag with the `git config` command. For example, to remove the user email from the global configuration:
```
